express-notes-api/server.js

```diff
@@ -0,0 +1,171 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const notesArray = [];
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+    for (const key in notes) {
+      notesArray.push(notes[key]);
+    }
+    if (notes) {
+      res.json(notesArray);
+    }
+  } catch (error) {
+    res.json([]);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const errorResponse = {};
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+
+    if (parseInt(req.params.id) < 0) {
```

Comment:
You parse the id 3 times. Why not just do it once and store it in a variable? That saves typing and means you don't have to remember to parse it if you add another place where you use the id.